packages/components/MenuButton/src/MenuButton.tsx

```diff
@@ -0,0 +1,96 @@
+/** @jsx withSlots */
+import React, {useRef, useState, useCallback} from 'react';
+import { Button } from '@fluentui-react-native/button';
+import { ContextualMenu, ContextualMenuItem } from '@fluentui-react-native/contextual-menu';
+import { IUseComposeStyling, compose } from '@uifabricshared/foundation-compose';
+import { mergeSettings } from '@uifabricshared/foundation-settings';
+import { ISlots, withSlots } from '@uifabricshared/foundation-composable';
+import { View } from 'react-native';
+
+import {
+  MenuButtonName,
+  MenuButtonProps,
+  MenuButtonSlotProps,
+  MenuButtonType,
+  MenuButtonRenderData,
+  MenuButtonState,
+} from './MenuButton.types';
+
+export const MenuButton = compose<MenuButtonType>({
+  displayName: MenuButtonName,
+  usePrepareProps: (userProps: MenuButtonProps, useStyling: IUseComposeStyling<MenuButtonType>) => {
+    const { menuItems, content, icon, disabled, onItemClick, contextualMenu } = userProps;
+
+    const stdBtnRef = useRef(null);
+    const [showContextualMenu, setShowContextualMenu] = useState(false);
+
+    const onDismiss = useCallback(() => {
+      setShowContextualMenu(false);
+    }, [setShowContextualMenu]);
+
+    const toggleShowContextualMenu = useCallback(() => {
+      setShowContextualMenu(!showContextualMenu);
+    }, [showContextualMenu, setShowContextualMenu]);
+
+    const state: MenuButtonState = {
+      context: {
+        showContextualMenu: !!showContextualMenu
+      },
+    };
+
+    const styleProps = useStyling(userProps, (override: string) => state[override] || userProps[override]);
+
+    const slotProps = mergeSettings<MenuButtonSlotProps>(styleProps, {
+      root: {},
+      button: {
+        content,
+        disabled,
+        icon,
+        componentRef: stdBtnRef,
+        onClick: toggleShowContextualMenu
+      },
+      contextualMenu: {
+        onItemClick,
+        target: stdBtnRef,
+        onDismiss,
+        setShowMenu: toggleShowContextualMenu,
+        ...contextualMenu
+      },
+      contextualMenuItems: {
+        menuItems
+      },
+      ContextualMenuItem
+    });
+
+    return { slotProps, state };
+  },
+  slots: {
+    root: React.Fragment,
+    button: { slotType: Button as React.ComponentType<object> },
+    contextualMenu: { slotType: ContextualMenu as React.ComponentType<object> },
+    contextualMenuItems: View
```

Comment:
hmm I'm not sure if View is the right slotType for the items. React.Fragment might be a better option since it's grouping the <ContextualMenuItem>s you are creating

Reply:
I agree. I tried with React.Fragment but I got an error. This slot was mostly for Submenu Items, because It has two elements, so I needed to wrap it
`
<>
  <SubmenuItem />
  <Submenu>
         <ContextualMenuItem />
  </Submenu>
</>
`

Reply:
Fixed